cfme/tests/cloud_infra_common/test_vm_instance_analysis.py

```diff
@@ -155,6 +155,35 @@ def _hosts_remove_creds():
             test_host.update_credentials_rest(credentials=Host.Credential(principal="", secret=""))
 
 
+def set_agent_creds(appliance, request, provider):
+    version = appliance.version.vstring
+    agent_data = {"ems": {"ems_amazon": {
+        "agent_coordinator": {"docker_image": "simaishi/amazon-ssa:{}".format(version),
+                              "docker_registry": "docker.io"}}}}
+    appliance.update_advanced_settings(agent_data)
+
+    # Adding SmartState Docker credentials
+    data = provider.data.endpoints
+    username = credentials[data['ssa_agent']['credentials']]['username']
+    password = credentials[data['ssa_agent']['credentials']]['password']
+    view = navigate_to(provider, "Edit")
+    ssa_agent_view = view.browser.create_view(EC2EndpointForm)
+    ssa_agent_view.smart_state_docker.select()
+    ssa_agent_view.smart_state_docker.username.fill(username)
+    ssa_agent_view.smart_state_docker.password.fill(password)
+    ssa_agent_view.default.validate.click()
+    view.save.click()
+
+    @request.addfinalizer
+    def _remove_docker_creds():
+        view = navigate_to(provider, "Edit")
+        ssa_agent_view = view.browser.create_view(EC2EndpointForm)
+        ssa_agent_view.smart_state_docker.select()
```

Comment:
Required:
ssa_agent_view.smart_state_docker.fill({
'username', ''
})

Reply:
okay will add it.